test/extended/testdata/builds/build-pruning/failed-pipeline.yaml

```diff
@@ -0,0 +1,86 @@
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "failed-pipeline"
+spec:
+  failedBuildsHistoryLimit: 2
+  strategy:
+    jenkinsPipelineStrategy:
+      jenkinsfile: |-
```

Comment:
the failed test could be a lot simpler, with less text ... perhaps sleep for x amount of time (where x is still pretty short), then throw an exception or make an obvious groovy invalid field reference error or something that will cause the pipeline to abort.

Reply:
yeah and just run it on the master, don't spin up slave pods.

probably doesn't even need to sleep.  just `sh 'exit 1'`